pkg/filesystem/fs.go

```diff
@@ -298,7 +300,14 @@ func (fs *Filesystem) Mount(snapshotID string, labels map[string]string) (err er
 			daemonconfig.CacheDir:  cacheDir,
 		}
 		cfg := deepcopy.Copy(fsManager.DaemonConfig).(daemonconfig.DaemonConfig)
-		err = daemonconfig.SupplementDaemonConfig(cfg, imageID, snapshotID, false, labels, params)
+		var updateErr error
+		err = daemonconfig.SupplementDaemonConfig(cfg, imageID, snapshotID, false, labels, params, func(imageHost string, keyChain *auth.PassKeyChain) {
+			logrus.Debugf("add key for %s", imageHost)
+			updateErr = fsManager.AuthCache.UpdateAuth(imageHost, keyChain.ToBase64())
+		})
+		if updateErr != nil {
```

Comment:
Passing closure's error code through an intermediate variable is a little tricky. How about give the closure a return value which can be returned by `SupplementDaemonConfig`

Reply:
Done.